spec/controllers/accounts/settings/webhooks_controller_spec.rb

```diff
@@ -0,0 +1,152 @@
+
+        context 'when it is an unauthenticated user' do
+        it 'returns unauthorized' do
+            post "/accounts/#{account.id}/webhooks" 
+            expect(response).to redirect_to(new_user_session_path)
+        end
+        end
+
+        context 'when it is an authenticated user' do
+        before do
+            sign_in(user)
+        end
+
+        context 'create webhook' do
+            it do
+            expect do
+                post "/accounts/#{account.id}/webhooks",
+                params: valid_params
+            end.to change(Webhook, :count).by(1)
+            expect(response).to redirect_to(account_webhooks_path(account))
+            end
+            context 'when url is invalid' do
+                it 'when url is blank' do
+                invalid_params = { webhook: { url: ''}}
+                expect do
+                    post "/accounts/#{account.id}/webhooks",
+                    params: invalid_params
+                end.to change(Webhook, :count).by(0)
+                expect(response.body).to include("Url não pode ficar em branco")
+                expect(response).to have_http_status(:unprocessable_entity)
+                end
+            end
+            context 'when status is invalid' do
+                it 'when status is blank' do
+                invalid_params = { webhook: { url: ''}}
+                expect do
+                    post "/accounts/#{account.id}/webhooks",
+                    params: invalid_params
+                end.to change(Webhook, :count).by(0)
+                expect(response.body).to include("Status não pode ficar em branco")
+                expect(response).to have_http_status(:unprocessable_entity)
+                end
+            end
+        end
+        end
+    end
+    describe 'GET /accounts/{account.id}/webhooks' do
+
+        context 'when it is an unauthenticated user' do
+        it 'returns unauthorized' do
+            get "/accounts/#{account.id}/webhooks" 
+            expect(response).to redirect_to(new_user_session_path)
+        end
+        end
+
+        context 'when it is an authenticated user' do
+        before do
+            sign_in(user)
+        end
+
+        context 'get webhooks' do
+            it do
+                get "/accounts/#{account.id}/webhooks"
+                expect(response.body).to include("https://woofedcrm.com")
+                expect(response).to have_http_status(200)
+            end
+            it 'get webhooks by account' do
+            account_2 = create(:account, name: 'account teste')
+            create(:webhook, url: 'teste-url.com.br', account_id: account_2.id)
+            get "/accounts/#{account.id}/webhooks"
+            expect(response.body).to include("https://woofedcrm.com")
+            expect(account.webhooks.count).to eq(1)
+            end
+        end
+        end
+    end
+    describe 'PACTH /accounts/{account.id}/users/{webhook.id}' do
```

Comment:
Daria pra adicionar um aqui de tentar editar um webhook de outra account

Reply:
Criei esse novo teste: 
![image](https://github.com/douglara/woofed-crm/assets/45243355/854d1c85-379e-4a60-bef6-b62f90015f08)
Mas ele está dando erro:
![image](https://github.com/douglara/woofed-crm/assets/45243355/3f4d2100-3faf-4f6d-aa5b-c98c5e3dc356)
Esse erro é de se esperar, pois ele está procurando um webhook que pertence a uma account_2 e ele n está encontrando e logo o sistema n deixa ele editar, o q é o correto. Mas n sei como eu faço para ele passar no teste sem ocorrer esse erro. @douglara

Reply:
Boaa isso mesmo!

Segue um exemplo de teste que da erro:
spec/integration/use_cases/accounts/apps/chatwoots/sync_inboxes_spec.rb:24

https://rspec.info/features/3-12/rspec-expectations/built-in-matchers/raise-error/